data_pipeline/utils/pipeline_config.py

```diff
@@ -26,40 +38,111 @@ class PipelineEnvironmentVariables:
 SECRET_VALUE_PLACEHOLDER = '***'
 
 
-class BigQuerySourceConfig(NamedTuple):
+@dataclass(frozen=True)
+class BigQuerySourceConfig:
     project_name: str
     sql_query: str
     ignore_not_found: bool = False
 
     @staticmethod
-    def from_dict(source_config_dict: dict) -> 'BigQuerySourceConfig':
+    def from_dict(source_config_dict: BigQuerySourceConfigDict) -> 'BigQuerySourceConfig':
         return BigQuerySourceConfig(
             project_name=source_config_dict['projectName'],
             sql_query=source_config_dict['sqlQuery'],
             ignore_not_found=source_config_dict.get('ignoreNotFound', False)
         )
 
 
-class BigQueryTargetConfig(NamedTuple):
+@dataclass(frozen=True)
```

Comment:
You raised why we are converting from `NamedTuple` to `dataclass`.
My reasoning was that some things, like extending base classes, can't be done with `NamedTuple`.
That is why I am tempted to convert all usage to `dataclass`.